### DISCUSSION TITLE
How to Build Planner Files for Windows Users…Step-by-Step

### DISCUSSION THREAD

**Discussion by mvgambino** [kudrykv/latex-yearly-planner#94] [Show and tell]
_*Edited this post to replace all references to "pdflatex" with "xelatex." As another user noted in a different post, using the "xelatex" command produced significantly smaller file sizes.*_

The planner that is available on this site is my absolute favorite. I love it so much that I know I’ll want a new version every year for the rest of my life! (Over-dramatic? I think not. 😊) The owner of this repository has been so gracious as to build new versions on demand each year, but I think the goal is that we should be able to customize & build our own. Unfortunately for me, I do not use the same system that the repository is primarily designed for, I use Windows 11. So most of the instructions on this site don’t work.

I read through all of the discussion posts and after lots of trial & error I’m happy to say that I have successfully cracked the code (pun intended) for Windows users! I sincerely hope that the step-by-step instructions below will help anyone successfully produce a customized PDF planner of their choosing.

My deepest gratitude to the producers & contributors that make this possible!

I’ve created a ridiculously long list of detailed steps so hopefully anybody can be successful, but at the end of the day there are really only 4 steps:
1. Install the required software & code repository
2. Choose your base planner style & identify the matching line of code you’ll need to use
3. Modify options in configuration files
4. Run the code to produce the PDF file

### Install Software & Copy the Code Repository

1.  Download & [Install LaTex (MiKTeX)](https://miktex.org/howto/install-miktex)
a.	After installing, apply updates using MiKTeX Console

2.	Download & [Install Go](https://go.dev/doc/install)
        a. Verify Go was installed properly
    1.	In Windows, click the Start menu.
    1.	In the menu's search box, type `cmd`, then press the Enter key.
    1.	In the Command Prompt window that appears, type the following command:  “`go version`” (without quotes)
    1.	Confirm that the command prints the installed version of Go.

3. Copy Code Repository from GitHub to your PC
a.	Download ZIP file from Code menu on GitHub
b.	Extract files to your PC
c.	Note the file path to the “latex-yearly-planner” folder
(Be prepared to copy/paste this path in the production steps)
### Identify Your Base Planner Style

4.	Decide which planner base configuration you want to build
a.	Device: Supernote, Remarkable, DDVK
b.	Navigation Style: Breadcrumb at Top v. Months on the Side (MOS)
c.	With/Without Daily Calendar: Mini Calendar on Daily Pages

6.	Identify the appropriate command line string for your desired planner base configuration:
a.	Open Windows File Explorer and navigate to the “latex-yearly-planner” folder (from step 3c)
b.	Open the “Release.sh” file using Windows Notepad
c.	Look for the Configurations section: “_configurations=(”
     1.	Note there are 17 different base configuration combinations as of the time of this post. (v0.0.8.2324)
_Eg: `1  "cfg/base.yaml,cfg/template_breadcrumb.yaml,cfg/sn_a5x.breadcrumb.default.yaml”   "sn_a5x.breadcrumb.default"`_
     1.	The first element identifies the number of times you’ll need to run the “xelatex” command. (1 or 2 times…you’ll use this when running the build code.)
     1.	The second element is the string of text you’ll need to pass into the “go” command to produce the LaTeX source files (tex files). 
_Eg: `"cfg/base.yaml,cfg/template_breadcrumb.yaml,cfg/sn_a5x.breadcrumb.default.yaml”`_
Be prepared to copy/paste this section of the string  in the production steps below.
           1.	Note: The comma-separated “cfg” values identify specific “.yaml” files in the CFG directory that are layered onto each other, from left-to-right, to produce the custom planner combination. The file to the right “wins” in terms of the configuration options selected within the files.
     1.	The third element identifies the PDF filename that will be produced. This is also part of the string that will need to be passed to the “xelatex” command.
_Eg: `sn_a5x.breadcrumb.default`_
     (Be prepared to copy/paste this string in the production steps)
    
     d.	Depending on your choices in Step 4, find the configuration sequence in the "release.sh" file (5, c , iii) that matches your preferences
     1.	My favorite is Months on the Side (MOS), Supernote, with Daily Calendar
     1.	My base configuration matches the following:
_`"cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml"`_
          1. This includes 4 .yaml files:
                     1.	Base file
                     3.	Template for MOS
                     4.	Supernote (sn_a5x) MOS
                     5.	Supernote Daily Calendar
     1.	Note that the number “2” precedes this configuration in the “Release.sh” code file. This means I’ll need to run the “xelatex” command 2 times. (*All of the MOS configurations require 2 passes*)
### Modify the Options You Want Applied to Your Planner

7.	Determine which features you want customized
a.	Lines v. Dots
b.	AM/PM v. 24 hour
c.	1st Day of Week (Sunday, Monday, etc.)
d.	Start/End Times on Daily Schedule

8.	Modify CFG code elements to suit your preferences
a.	Open the “latex-yearly-planner/cfg” folder
b.	Based on your configuration selection (from Step 5, c, iii,), open each of the “.yaml” files, one at a time, from left to right and look for the following options. (Some options will appear in multiple .yaml files, this is why it’s important to know that the file to the right “wins”)
     1.	With/Without Mini Calendar on Daily Schedule Page
          1)	`calafterschedule: true` (true = with calendar,  false = without calendar)
     1.	Start/End Times on Daily Schedule
           1)	`dailybottomhour: 8` (choose an hour 1 – 24)
           2)	`dailytophour: 19` (choose an hour 1 – 24)
     1.	1st Day of Week
          1)	`weekstart: 1` (0 = Sunday – 6 = Saturday)
     1.	AM/PM v. 24 hour
          1)	`ampmtime: true` (true = am/pm, false = 24 hour)
     1.	Lines v. Dots
          1)	`dotted: false` (true = dots, false = lines)

      c.	Note – Depending on the choices made for the mini calendar and start/end times, you may need to exclude the line spacing for the last half hour on the daily schedule page. (In some cases, one extra line can push the mini calendar off the page causing formatting issues in your final PDF) If necessary look for this option:
     1.	`addlasthalfhour: true` (true or false)
### Produce Your Planner!

9.	Run the Code to Build Your Planner
a.	Open Windows Command Prompt
     1.	Click on the Start Menu
     1.	Type “`cmd`”
     
     b.	Navigate to the “latex-yearly-planner” directory from step 3c
     1.	I use Windows File Explorer to navigate to the folder, then I copy the path from the Address Bar
     1.	In the Command Prompt (after the “>” symbol), type:“`cd ` “ (note there is a space after “cd”) Do not press enter yet…
     1.	Now paste the path to your “latex-yearly-planner” folder
_Eg: >`cd C:\Users\username\Documents\GitHub\latex-yearly-planner`_
     1.	After pasting the path, press Enter. Your command prompt should now reflect that you are in the correct folder:
_Eg: C:\Users\username\Documents\GitHub\latex-yearly-planner>_

     c.	Set the Year for your planner file (2025 in this example)
     1.	Type the following at the Command Prompt (after the “>” symbol):
`set PLANNER_YEAR=2025`
     1.	Press Enter
     
     d.	Run the command to build the “tex” files for your planner
     1.	Copy/Paste the following at the command prompt (don’t hit Enter yet):
`go run cmd/plannergen/plannergen.go --config ` 
     1.	Now Copy/Paste the configuration string of the planner type you want (See step 5d). You’ll need to include the quote symbols. My personal favorite is:
`"cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml"`
     1.	Altogether, the Command Prompt would now show:
`
go run cmd/plannergen/plannergen.go --config 
"cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml"
`
(This should be one continuous command line, however this post creates the appearance of a line break. See the "Tips and Tricks" section below for an actual, scrollable view.)
     1.	Press Enter
          1)	There will likely be a momentary pause before the command prompt reappears
          2)	You’ll know the command operated successfully if there are a lot of new files in the “out” folder in the “latex-yearly-planner” directory
          3)	These files will be used by MiKTeX to create the PDF file in the next step

     e.	Change to the “Out” folder before running the next command(s)
     1.	Type the following at the Command Prompt (after the “>” symbol):
`cd out`
     1.	Press Enter. Your command prompt should now reflect that you are in the correct folder:
_C:\Users\username\Documents\GitHub\latex-yearly-planner\out>_

     f.	Run the command to build the PDF planner file
     1.	Copy/Paste the following at the command prompt (don’t hit Enter yet):
`xelatex`
     1.	Now Copy/Paste the filename string for the planner type you want (See step 5,c,iv). DO NOT include the quote symbols. The example for my personal favorite is:
`sn_a5x.mos.default.dailycal`
     1.	Add “`.tex`” at the end of the filename
     1.	Altogether, the Command Prompt would now show:
_`xelatex sn_a5x.mos.default.dailycal.tex`_
     1.	Press Enter
          1)	If this is the 1st time you’ve run this command you may be presented with several questions about installing updates…accept all of them. (I believe there is an option that allows you to accept all instead of accepting 1-at-a-time.)
          2)	It will take some time for this process to complete and you will see a lot of text showing the progress…just wait it out, the result is worth it!
     1.	If you selected a Breadcrumb planner, your file is complete!
     1.	If you selected a Months on the Side (MOS) planner, repeat step 8f
          1)	The 1st pass created the .tex files necessary for layering
          2)	The 2nd pass completes the layering process
          3)	Your file is complete!

     g.	Look for the PDF file for the planner of your dreams in the “out” folder
     1.	There are a LOT of files in the “out” folder, but one of them should be your PDF planner!

### Tips and Tricks

1. Once you identify the elements of the planner you love and you update all of the options you want in the .yaml files, you can compile a quick-access file containing all of the Windows Command line entries you’ll need to create future files (just change the date). For me, I created a .txt file with the following content:

```
CD C:\Users\username\Documents\GitHub\latex-yearly-planner
set PLANNER_YEAR=2025
go run cmd/plannergen/plannergen.go --config "cfg/base.yaml,cfg/template_months_on_side.yaml,cfg/sn_a5x.mos.default.yaml,cfg/sn_a5x.mos.default.dailycal.yaml"
CD out
xelatex sn_a5x.mos.default.dailycal.tex
xelatex sn_a5x.mos.default.dailycal.tex
```

Now I can simply run the Windows Command prompt, then copy/paste each command in sequence to build additional planner files.

Since I'm not a "coder," I'm sure I've over-complicated some things, but all I know is that these steps WORKED for me & I'm hopeful that I have helped others! :)

Cheers!

**Comment by kudrykv** [comment]
Thank you for this guide!

**Reply by mvgambino** [reply]
Thanks for producing the planner! I just updated the instructions to replace references to using the "pdflatex" command with the "xelatex" command. As another user discovered, it really does result in significantly smaller file sizes. (~3Mb vs. 11-12Mb). I love this planner so much!

**Comment by sooyhwang** [comment]
This is awesome! Also, [this discussion post](https://github.com/kudrykv/latex-yearly-planner/discussions/93) has tips on using different latex s/w for smaller file size.

**Reply by mvgambino** [reply]
Thank you, I edited the instructions to use the "xelatex" command instead of the "pdflatex" command. It does indeed produce smaller file sizes. :)

**Comment by poloman172** [comment]
mvgambino - thanks for the guide - I'm going to give it a go. 

I use the planner on an iPad in Noteability but it slows down after a few months worth of use - currently i chop up the PDF in Adobe but it takes a while each year so will see how i get on with compiling using the code and your guide - thanks for all the effort.

I'm also going to try to get the ToDo back in there so will learn a bit about github and past versions. 2024 - all about the Learning! :)

**Comment by endsouza** [comment]
Thank you! I tested on a Mac M1 and it worked like charm!

**Reply by mvgambino** [reply]
I'm so glad the instructions worked for you! I just edited the instructions to replace using the "pdflatex" command with the "xelatex" command. The xelatex command produces much smaller file sizes (~3Mb).

**Reply by EZ707** [reply]
> Thank you! I tested on a Mac M1 and it worked like charm!

May I ask you, how you managed set year? variables not changing to me, I am also doing on m1

**Comment by edanguilan** [comment]
Thanks! with a little bit research and debugging, i finaly made this work! i have set my 2025 planner in am pm. THANKS YOU SO MUCH!

**Comment by enmas23** [comment]
Thank you for the guide. I need your help: no matter what I do, I always get 2024 planner. I set the year as you wrote, but nothing. Can you help me, please? Thank you very much!

**Comment by edanguilan** [comment]


**Reply by enmas23** [reply]
The image is not visible, sorry

**Comment by edanguilan** [comment]
Look at the guide.... the part where it says setdate=2025

**Reply by enmas23** [reply]
yes, set PLANNER_YEAR=2025
I do this, but I always get a 2024 planner

**Reply by mvgambino** [reply]
I'm not sure where your problem lies. Perhaps you haven't switched to the appropriate directory before typing the "Set PLANNER_YEAR" command? The command needs to be run while in the "latex-yearly-planner" folder.

**Reply by EZ707** [reply]
did you solved this issue? I have same, year stucked on 2024

**Reply by enmas23** [reply]
Kind of. I opened the terminal as administrator from inside the shell and it worked. It seems like you can not update the variables if you aren't in administrator mode

**Reply by EZ707** [reply]
thank you, I am on Mac m1, cant do same.

**Reply by stalegjelsten** [reply]
> thank you, I am on Mac m1, cant do same.

@EZ707 on a Mac you should try typing `PLANNER_YEAR=2025` without the preceding `set`.

**Reply by EZ707** [reply]
> > thank you, I am on Mac m1, cant do same.
> 
> @EZ707 on a Mac you should try typing `PLANNER_YEAR=2025` without the preceding `set`.

appreciated, thank you

**Comment by YasnoSolnishko** [comment]
Is there an option to remove the schedule?
So all area would be filled with "notes" and "my priority."

**Comment by edanguilan** [comment]
I tried to make another planner but this time I can not set the planner year... dunno what to do now. It was working before.

**Reply by edanguilan** [reply]
im using windows OS

**Comment by ricarm3** [comment]
Hi everyone... I followed the steps outlined here, and while the process seems to work, the output pdf file is messed up. I think its just spacing issues. On the log file after running the xelatex i get the following error:

The control sequence at the end of the top line
of your error message was never \def'ed. If you have
misspelled it (e.g., `\hobx'), type `I' and the correct
spelling (e.g., `I\hbox'). Otherwise just continue,
and I'll forget about whatever was undefined.

! Undefined control sequence.
<argument> ...rtpbox {\@nextchar }\insert@pcolumn 
                                                  \@endpbox \do@row@strut \o...
l.22086 \medskip


BUT, I don't know exactly how to resolve this....any help is appreciated


<img width="733" height="937" alt="Screenshot 2025-07-18 133412" src="https://github.com/user-attachments/assets/da41131c-26ff-4656-a8aa-16fc4dea1669" />
<img width="712" height="547" alt="Screenshot 2025-07-18 133438" src="https://github.com/user-attachments/assets/e3a0e4a9-c967-4a96-b08a-10b9b8cb71fc" />
<img width="692" height="1118" alt="Screenshot 2025-07-18 133507" src="https://github.com/user-attachments/assets/fe2aa7e5-aa5d-495b-88b2-bec0109f0728" />
